### DISCUSSION TITLE
Hooks

### DISCUSSION THREAD

**Discussion by jghauser** [papis/papis#931] [Q&A] [answered]
Does anyone know what's the reason for the `on_edit_done` hook being triggered *before* updating the db? I've had the idea of automating exports to my `.bib` file after each edit of a `info.yaml`, but I think that wouldn't work (reliably) because of that. Or am I wrong in this assumption?

**Comment by alexfikl** [comment, marked as answer]
I conveniently moved that in #892 :grin: My reasoning was that it could be used to perform some final touch ups to an edited document before saving it to the database. If we called it after updating the db, then we would need to save it again in case the hook made additional edits (?).

How are you trying to use the hooks for exporting?

**Reply by jghauser** [reply]
Haha! I didn't really think about it much yet, but the idea was to have a hook that does something like `papis export --all --out ~/main.bib '*'` whenever I use `papis edit` to change something. Though on second thought it would be a little brittle anyway as it wouldn't catch other kinds of edits (e.g. with `papis update`).

**Reply by alexfikl** [reply]
Yeah, the hook stuff is not very developed (just added in a couple of places), so I'm not sure it's worth relying on it too much unless something really can't be done in any other way :(

**Reply by jghauser** [reply]
Yeah that makes sense. It's not actually something that comes up a lot (I rarely edit docs after adding them). And on second second thought, I mostly do that from `papis.nvim` anyway, so I might add a hook functionality there :). Thanks for the quick response!
